### PR TITLE
Replace Set of AttributeMatcher with the List

### PR DESCRIPTION
**Pull request by robsunday**

### CODE REVIEW

File: jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/assertions/DataPointAttributes.java [hunk 1]
```diff
@@ -40,21 +42,30 @@ public static AttributeMatcher attributeWithAnyValue(String name) {
...
-   *     used for matching
+   * @see MetricAssert#hasDataPointsWithAttributes(Collection[]) for detailed description of the
+   *     algorithm used for matching
    */
-  public static Set<AttributeMatcher> attributeSet(AttributeMatcher... attributes) {
-    Set<AttributeMatcher> matcherSet = Arrays.stream(attributes).collect(Collectors.toSet());
-    if (matcherSet.size() < attributes.length) {
-      throw new IllegalArgumentException(
-          "Duplicated matchers found in " + Arrays.toString(attributes));
+  public static List<AttributeMatcher> attributeSet(AttributeMatcher... attributes) {
```
Comment (robsunday) [inline jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/assertions/DataPointAttributes.java:56]: [for reviewer] What do you think about renaming this method to attributeGroup ? The current one is somehow suggesting that Set is returned

File: jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/assertions/MetricAssert.java [hunk 2]
```diff
@@ -293,12 +293,17 @@ public final MetricAssert hasDataPointsWithAttributes(
   }
 
   private static boolean matchAttributes(
-      Set<AttributeMatcher> attributeMatchers, Map<String, String> dataPointAttributes) {
+      Collection<AttributeMatcher> attributeMatchers, Map<String, String> dataPointAttributes) {
```
Comment (robsunday) [inline jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/assertions/MetricAssert.java:296]: [for reviewer] Using Collection is more future-proof, but I can change it to List

File: jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/assertions/MetricAssert.java [hunk 3]
```diff
@@ -293,12 +293,17 @@ public final MetricAssert hasDataPointsWithAttributes(
...
     if (attributeMatchers.size() != dataPointAttributes.size()) {
       return false;
     }
+
     for (AttributeMatcher matcher : attributeMatchers) {
-      String attributeValue = dataPointAttributes.get(matcher.getAttributeName());
+      String attributeName = matcher.getAttributeName();
```
Comment (robsunday) [inline jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/assertions/MetricAssert.java:302]: [for reviewer] This covers missing edge case

File: jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/assertions/AttributeMatcher.java [hunk 4]
```diff
@@ -29,6 +29,7 @@ public class AttributeMatcher {
    * @param attributeValue attribute value
    */
   AttributeMatcher(String attributeName, @Nullable String attributeValue) {
+    Objects.requireNonNull(attributeName, "Null attribute name");
```
Comment (robsunday) [inline jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/assertions/AttributeMatcher.java:32]: [for reviewer] Just additional early check

File: jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/assertions/DataPointAttributes.java (hunk 1 above)
Comment (SylvainJuge) [reply jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/assertions/DataPointAttributes.java:56]: +1 on this, it even removes the ambiguity with sets/collections.

File: jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/assertions/MetricAssert.java (hunk 2 above)
Comment (SylvainJuge) [reply jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/assertions/MetricAssert.java:296]: The problem with having a generic type in var-arg is that it likely adds another warning, this is probably where using an intermediate collection class can help.

**Pull request closed without merging**